### PR TITLE
Add changes to DG and diagrams

### PR DESCRIPTION
**Pull request by wilsonwid**
This PR adds changes to the developer guide (DG) and related diagrams.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -71,20 +71,20 @@ The **API** of this component is specified in [`Ui.java`](https://github.com/se-
...
+The `Ui` component uses the JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/AY2324S2-CS2103T-W08-3/tp/blob/master/src/main/java/scm/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/AY2324S2-CS2103T-W08-3/tp/blob/master/src/main/resources/view/MainWindow.fxml)
 
-The `UI` component,
+The `Ui` component,
 
 * executes user commands using the `Logic` component.
 * listens for changes to `Model` data so that the UI can be updated with the modified data.
-* keeps a reference to the `Logic` component, because the `UI` relies on the `Logic` to execute commands.
-* depends on some classes in the `Model` component, as it displays `Person` object residing in the `Model`.
+* keeps a reference to the `Logic` component, because the `Ui` relies on the `Logic` to execute commands.
```
Comment (sarjinius) [inline docs/DeveloperGuide.md:82]: "`Ui` component relies on the `Logic` component" would be better here.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -71,20 +71,20 @@ The **API** of this component is specified in [`Ui.java`](https://github.com/se-
...
+* depends on some classes in the `Model` component, as it displays `Person` and `Schedule` objects residing in the `Model`.
```
Comment (sarjinius) [inline docs/DeveloperGuide.md:83]: "as it displays **the** `Person` and `Schedule` objects" would be better here.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -116,22 +116,22 @@ How the parsing works:
...
+**API** : [`Model.java`](https://github.com/AY2324S2-CS2103T-W08-3/tp/blob/master/src/main/java/scm/address/model/Model.java)
 
 <puml src="diagrams/ModelClassDiagram.puml" width="450" />
 
 
 The `Model` component,
 
-* stores the contact manager data i.e., all `Person` objects (which are contained in a `UniquePersonList` object), as well as all schedule list data (i.e., all `Schedule` objects).)
+* stores the contact manager data, i.e., all `Person` objects (which are contained in a `UniquePersonList` object), as well as all schedule list data (i.e., all `Schedule` objects, which are contained in a `ScheduleList` object).
 * stores the currently 'selected' `Person` objects (e.g., results of a search query) as a separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
```
Comment (sarjinius) [inline docs/DeveloperGuide.md:127]: "that can be observed (insert comma) e.g. (insert comma) the UI"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -116,22 +116,22 @@ How the parsing works:
...
 * stores the currently 'selected' `Schedule` objects in a manner analogous to the above 'selected' `Person` objects.
 * stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a `ReadOnlyUserPref` objects.
 * does not depend on any of the other three components (as the `Model` represents data entities of the domain, they should make sense on their own without depending on other components)
 
 <box type="info" seamless>
 
-**Note:** An alternative (arguably, a more OOP) model is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique tag, instead of each `Person` needing their own `Tag` objects.<br>
+**Note:** An alternative (arguably, a more OOP) model for the `AddressBook` part of `Model` is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique tag, instead of each `Person` needing their own `Tag` objects.<br>
```
Comment (sarjinius) [inline docs/DeveloperGuide.md:134]: "(arguably, a more OOP-**oriented**) model for the `AddressBook` part of **the** `Model` **component**"

Comment (sarjinius) [reply docs/DeveloperGuide.md:134]: "which **a** `Person` references"

**Review by sarjinius — CHANGES_REQUESTED** [review]
LGTM except for minor changes.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (wilsonwid) [reply docs/DeveloperGuide.md:134]: I don't think this requires changes.

**Review by sarjinius — APPROVED** [review]

**Pull request merged**